tests/Aspire.Hosting.Tests/ManifestGenerationTests.cs

```diff
@@ -183,6 +183,43 @@ public void EnsureContainerWithArgsEmitsContainerArgs()
             arg => Assert.Equal("more", arg.GetString()));
     }
 
+    [Fact]
+    public void EnsureContainerWithVolumesEmitsVolumes()
+    {
+        using var program = CreateTestProgramJsonDocumentManifestPublisher();
+
+        program.AppBuilder.AddContainer("containerwithvolumes", "image/name")
+                          .WithVolume("myvolume", "/mount/here")
+                          .WithBindMount("./some/source", "/bound") // This should be ignored and not written to the manifest
+                          .WithVolume("myreadonlyvolume", "/mount/there", isReadOnly: true)
+                          .WithVolume(null! /* anonymous volume */, "/mount/everywhere");
+
+        // Build AppHost so that publisher can be resolved.
+        program.Build();
+        var publisher = program.GetManifestPublisher();
+
+        program.Run();
+
+        var resources = publisher.ManifestDocument.RootElement.GetProperty("resources");
```

Comment:
We have a new way of checking the manifest structure. Use `ManifestUtils.GetManifest(resource)` then have a big `"""{json}"""` string which you compare with the `ToString` method above. There are some tests in there that already do this.

The benefit of this approach is it helps identify manifest changes we aren't specifically looking for.

Reply:
... its also a lot less Asserts for you to keep track of :)